src/joystick/hidapi/SDL_hidapijoystick.c

```diff
@@ -600,6 +733,25 @@ HIDAPI_JoystickInit(void)
         return 0;
     }
 
+#if defined(SDL_USE_LIBUDEV)
+    if (linux_enumeration_method == ENUMERATION_UNSET) {
+        if (SDL_getenv("SDL_HIDAPI_JOYSTICK_DISABLE_UDEV") != NULL) {
+            SDL_LogDebug(SDL_LOG_CATEGORY_INPUT,
+                         "udev disabled by SDL_HIDAPI_JOYSTICK_DISABLE_UDEV");
+            linux_enumeration_method = ENUMERATION_FALLBACK;
+        } else if (access("/.flatpak-info", F_OK) == 0
+                   || access("/run/pressure-vessel", F_OK) == 0) {
```

Comment:
While you're here, it might be good to also probe `/run/host/container-manager` (here and in the evdev code), which [is container-manager-agnostic](https://systemd.io/CONTAINER_INTERFACE/).

Checking for `/run/host/container-manager` would in principle make it unnecessary to check for `/run/pressure-vessel` (all supported versions of pressure-vessel also have `/run/host/container-manager`), but checking `/.flatpak-info` will still be necessary, because older distributions have older versions of Flatpak that didn't implement `/run/host/container-manager`.

Reply:
Implemented in the latest revision.